### PR TITLE
@kanaabe => New auth modal system

### PR DESCRIPTION
**Pull request by l2succes**
This PR replaces the existing authentication modal system for new react based components.

### CODE REVIEW

**Comment by kanaabe** [comment]
PR'd the latest here: https://github.com/l2succes/force/pull/2 -- I'll probably start pushing straight to this branch after this PR since but lmk if you prefer the former.

**Comment by xtina-starr** [comment]
Remaining modal work:
- [ ] Verify static auth and mediator generated modals match existing functionality and behave as expected.
- [ ] Get this PR merged into a heroku QA environment.
- [ ] Start working on the remaining auth modal (e.g. Artist Page CTA, marketing signup modal, embedded sign form, professional buyers).

File: src/desktop/apps/auth2/routes.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  try {
+    const layout = await renderLayout({
+      basePath: __dirname,
+      layout: '../../components/main_layout/templates/react_blank_index.jade',
+      config: {
+        styledComponents: true,
+      },
+      blocks: {
+        head: 'templates/meta.jade',
```
Comment (damassi) [inline src/desktop/apps/auth2/routes.js]: Sorry for always pinging on line of code, but can this be replaced with a js file and a `<Fragment />`? The less jade we use in new code the better, and with React's Fragment support using `.jade` files in the head is no longer necessary.

Comment (xtina-starr) [reply src/desktop/apps/auth2/routes.js]: no need to apologize. the gentle nudge is the only way it's gonna stick. i'll address it in the next commit.

**Comment by xtina-starr** [comment]
This [PR](https://github.com/l2succes/force/pull/3) ensures the new auth modal is triggered via the mediator.

**Comment by l2succes** [comment]
Updated the PR with the submission and redirection work which depends on https://github.com/artsy/reaction/pull/710 to work. Still, need to add rendering errors that come back from the backend.

File: src/desktop/apps/auth2/client/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    return (
+      <ModalManager
+        ref={ref => (manager = ref)}
+        submitUrls={{
+          login: sd.AP.loginPagePath,
+          signup: sd.AP.signupPagePath,
+        }}
+        csrf={sd.CSRF_TOKEN}
+        redirectUrl="/"
```
Comment (l2succes) [inline src/desktop/apps/auth2/client/index.tsx]: should pass in the `redirect-to` property from the mediator here and default to "/"

**Comment by l2succes** [comment]
TODO

- [ ] Render backend errors (e.g. "email or password incorrect")
- [ ] Add a loading state while the form is submitting
- [ ] Add show/hide animations to modals

**Pull request merged**